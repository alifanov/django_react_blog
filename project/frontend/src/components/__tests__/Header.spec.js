import React from "react";
import Header, { Header as DummyHeader } from "../Header";
import { shallow, mount } from "../../enzyme";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";

import renderer from "react-test-renderer";

const mockStore = configureStore();

const store = mockStore({
  auth: {
    access: { token: "token" },
    refresh: {
      token: "token"
    }
  }
});

describe(">>> Header test", () => {
  let header;
  beforeEach(() => {
    header = shallow(<Header store={store} />);
  });

  it("+++ render Header component", () => {
    expect(header.length).toEqual(1);
  });
  it("+++ check text for Header", () => {
    const header = shallow(<DummyHeader isAuthenticated={false} />);
    expect(header.find("h5").text()).toEqual("Simple blog");
  });
});

describe(">>>Header --- Snapshot", () => {
  it("+++capturing Snapshot of Header", () => {
    const renderedValue = renderer
      .create(
        <MemoryRouter>
          <Header store={store} />
        </MemoryRouter>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
