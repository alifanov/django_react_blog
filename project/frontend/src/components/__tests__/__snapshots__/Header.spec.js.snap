// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Header --- Snapshot +++capturing Snapshot of Header 1`] = `
<div
  className="d-flex flex-column flex-md-row align-items-center p-3 px-md-4 mb-3 bg-white border-bottom shadow-sm"
>
  <h5
    className="my-0 mr-md-auto font-weight-normal"
  >
    Simple blog
  </h5>
  <nav
    className="my-2 my-md-0 mr-md-3"
  >
    <a
      className="p-2 text-dark"
      href="/posts/"
      onClick={[Function]}
    >
      Posts
    </a>
    <a
      className="p-2 text-dark"
      href="/tags/"
      onClick={[Function]}
    >
      Tags
    </a>
    <a
      className="p-2 text-dark"
      href="#"
    >
      Page #3
    </a>
    <a
      className="p-2 text-dark"
      href="#"
    >
      Page #4
    </a>
  </nav>
  <a
    className="btn btn-outline-primary"
    href="#"
  >
    Sign up
  </a>
</div>
`;
